core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -174,6 +181,41 @@ URI uri() {
         }
     }
 
+    @Nullable
+    List<LanguageRange> acceptLanguages() {
+        final List<String> acceptHeaders = getAll(HttpHeaderNames.ACCEPT_LANGUAGE);
+        if (acceptHeaders.isEmpty()) {
+            return null;
+        }
+
+        try {
+            final List<LanguageRange> acceptLanguages = new ArrayList<>(4);
+            for (String acceptHeader : acceptHeaders) {
+                acceptLanguages.addAll(LanguageRange.parse(acceptHeader));
+            }
+            acceptLanguages.sort(comparingDouble(LanguageRange::getWeight).reversed());
+
+            return ImmutableList.copyOf(acceptLanguages);
+        } catch (IllegalArgumentException e) {
+            // If any port of any of the headers is ill-formed
+            return null;
+        }
+    }
+
+    @Nullable
+    Locale selectLocale(Collection<Locale> supportedLocales) {
+        requireNonNull(supportedLocales, "supportedLocales");
+        final List<LanguageRange> languageRanges = acceptLanguages();
+        if (languageRanges == null || supportedLocales.isEmpty()) {
```

Comment:
How about checking `supportedLocales` is empty before calling `acceptLanguages()`? If it's empty we don't need to parse the header.

Reply:
Will do

Reply:
I don't understand the following? If you mean I should add `Locale...`. If so, absolutely. It was just a cultural misunderstanding. I sincerely thought it was a question where @ikhoon wanted my opinion, not a light version of "please do it this way". :laughing: 

> How about adding now for consistency? :pray: You can use `ImmutableList.of()`.

Reply:
I'm not @ikhoon, so I'm not 100% sure, but I guess he thought he'll just add it later by himself since he is such a nice guy who doesn't want to waste your time doing a tedious job. :wink:

Reply:
I also somehow missed his reply >_<. Sorry ikhoon.

Reply:
> I also somehow missed his reply >_<. Sorry ikhoon.

No problem. 🤣🤣🤣